api/src/main/java/edu/wpi/first/shuffleboard/api/plugin/Plugin.java

```diff
@@ -0,0 +1,123 @@
+package edu.wpi.first.shuffleboard.api.plugin;
+
+import com.google.common.collect.ImmutableList;
+
+import edu.wpi.first.shuffleboard.api.data.DataType;
+import edu.wpi.first.shuffleboard.api.sources.SourceType;
+import edu.wpi.first.shuffleboard.api.sources.recording.serialization.TypeAdapter;
+import edu.wpi.first.shuffleboard.api.widget.Widget;
+
+import java.util.List;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+
+/**
+ * Common superclass for plugins that can be loaded by the app at startup or during runtime. Subclasses must have
+ * public no-arg constructor or they will not be loaded.
+ */
+public class Plugin {
```

Comment:
This makes it easy to create new instances for testing. There are also no abstract methods deliberately because implementations may implement any or none of the methods of this class